src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -68,10 +81,10 @@ internal static string NormalizeDirectorySeparators(string path)
                     && (i + 1 < path.Length && IsDirectorySeparator(path[i + 1])))
                     continue;
 
-                builder.Append(current);
+                sb.Append(current);
             }
 
-            return builder.ToString();
+            return result;
```

Comment:
The result will be always empty Span here.

Reply:
will sb.toString().AsReadOnlySpan() will serve our purpose ?

Reply:
It would be functionally correct, but there are quite a few unnecessary copies along the way - especially when this wrapped by the `string` overload. I think you may want:
- Rename this method to NormalizeDirectorySeparatorsIfNecessary
- Change its return type to string and make it return null if the directory separators are normalized already
- Use the `string Create<TState>(int length, TState state, SpanAction<char, TState> action)` method to construct the string
- Undo the change that added the new ValueStringBuilder constructor because of it won't be used.

Reply:
The argument should be ReadOnlySpan<char> or string ?

Reply:
ReadOnlySpan

Reply:
```         return string.Create(path.Length, (path, start), (dst, state) =>
            {
                int i = state.start;
                int j = 0;

                if (IsDirectorySeparator(state.path[state.start]))
                {
                    i++;
                    dst[j++] = DirectorySeparatorChar;
                }

                for (; i < path.Length; i++)
                {
                    current = path[i];

                    // If we have a separator
                    if (IsDirectorySeparator(current))
                    {
                        // If the next is a separator, skip adding this
                        if (i + 1 < path.Length && IsDirectorySeparator(path[i + 1]))
                        {
                            continue;
                        }

                        // Ensure it is the primary separator
                        current = DirectorySeparatorChar;
                    }
                    dst[j++] = current;
                }
            });
```

Reply:
I am trying to do it but its giving me an error that ReadOnlySpan<char> may not be used as a type argument. can you help me ?

Reply:
Hmm. You have to pin the string and use unsafe code unfortunately. E.g. similar to how it is done here: https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/PathHelpers.cs#L123